chain/consensus/mir/mempool/handlers/batches.go

```diff
@@ -21,54 +21,37 @@ func IncludeBatchCreation(
 	params *types.ModuleParams,
 	commonState *types.State,
 ) {
-	state := &State{
-		State:    commonState,
-		NewTxIDs: nil,
-	}
-
-	dsl.UponNewRequests(m, func(txs []*requestpb.Request) error {
-		mpdsl.RequestTransactionIDs(m, mc.Self, txs, &requestTxIDsContext{txs})
-		return nil
-	})
-
 	mpdsl.UponTransactionIDsResponse(m, func(txIDs []t.TxID, context *requestTxIDsContext) error {
-		for i := range txIDs {
-			state.TxByID[txIDs[i]] = context.txs[i]
-		}
-		state.NewTxIDs = append(state.NewTxIDs, txIDs...)
-		return nil
-	})
-
-	mpdsl.UponRequestBatch(m, func(origin *mempoolpb.RequestBatchOrigin) error {
-		var txIDs []t.TxID
 		var txs []*requestpb.Request
-		batchSize := 0
-
 		txCount := 0
-		for _, txID := range state.NewTxIDs {
-			tx := state.TxByID[txID]
+		for i := range txIDs {
+			tx := context.txs[i]
+			commonState.TxByID[txIDs[i]] = tx
 
 			// TODO: add other limitations (if any) here.
 			if txCount == params.MaxTransactionsInBatch {
```

Comment:
This check should probably not be enforced here any more. The handler should just take all transactions it receives and put them in a batch. The limit should be enforced when the transactions are fetched from the Eudico mempool.

Reply:
Fixed